### PR TITLE
[model] fix: apply attention mask in flux flash-attention path

### PR DESCRIPTION
**Pull request by rootkiller6788**
## Summary

Fix `flash_attention` dropping the attention mask on the flash-attention (GPU) path, so masked attention behaves consistently across backends.

## Problem

`flash_attention` applies `attn_mask` only in the CPU fallback branch (`F.scaled_dot_product_attention`). The flash-attention branches (FA2/FA3) call `flash_attn_func(q, k, v, causal=causal)` without passing the mask, so when the model uses `entity_masks` (the text-image conditioning mask built by `construct_mask`), the mask is silently ignored on GPU.

This makes GPU and CPU produce different attention behavior for the same masked inputs.

## Fix

When `attn_mask` is not `None`, fall back to `scaled_dot_product_attention`, which supports an arbitrary additive mask (note: supplying a mask rules out SDPA's flash backend, so this uses the memory-efficient kernel). The flash-attention fast path remains for the mask-less case.

The mask is a `(b, 1, s, s)` additive bias (`0` / `-inf`) built by `construct_mask`, which `scaled_dot_product_attention` accepts directly. When `causal=True`, a lower-triangular causal bias is merged into the additive mask first, since `scaled_dot_product_attention` cannot take both `attn_mask` and `is_causal` — this preserves the caller's causal intent instead of silently dropping it. A boolean mask is normalised to the additive `0`/`-inf` form first, since `bool + float` would otherwise promote to `float32` and corrupt the mask.

## Verification

- `ruff check` and `ruff format --check` pass.
- Verified numerically that a `(b, 1, s, s)` `0` / `-inf` mask is applied correctly: masked positions receive zero attention weight, and the result matches a manual `softmax(scores + mask) @ v` reference.
- Verified `causal=True` + float mask matches `softmax(scores + mask + causal_bias) @ v`, and `causal=True` + bool mask matches the same after normalisation (max abs diff ~1e-7).

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved attention handling when masks are supplied.
  * Added reliable causal masking support for boolean and additive masks.
  * Preserved existing optimized attention behavior for inputs without masks.

### CODE REVIEW

File: veomni/models/transformers/flux/modeling_flux.py [hunk 1]
```diff
@@ -83,6 +83,14 @@ def rearrange_qkv(q: torch.Tensor, k: torch.Tensor, v: torch.Tensor, rerange_typ
 def flash_attention(q: torch.Tensor, k: torch.Tensor, v: torch.Tensor, causal: bool = False, attn_mask=None):
     # bs, head_cont, seq, head_dim = q.shape
 
+    if attn_mask is not None:
+        # flash_attn_func (FA2/FA3) does not accept an arbitrary attention mask. The
+        # mask is a (b, 1, s, s) additive bias (0 / -inf) built by `construct_mask`, so
+        # fall back to scaled_dot_product_attention, which supports it and still
+        # dispatches to an efficient fused backend. The flash-attn path below is for the
+        # mask-less case only.
+        return F.scaled_dot_product_attention(q, k, v, attn_mask=attn_mask)
```
[the remark this answers is not in the document]
Comment (rootkiller6788) [reply veomni/models/transformers/flux/modeling_flux.py:101]: addressed in 41e3bec6. The SDPA fallback now merges a lower-triangular causal bias into the additive mask when
 causal=True, preserving the caller's causal intent. I'm happy to add a focused regression test if the maintainers would like one.

**Comment by rootkiller6788** [comment]
Thanks for the thorough review — all four points addressed in b14366c2:

  - **ruff format**: replaced the `torch.full((seq, seq), ...)` bias with `torch.full_like(attn_mask, ...)`, which
  shortens the line (no 119-char collapse) and also drops the square-shape assumption (`seq_q == seq_k`).
  - **bool mask**: added a normalisation guard — `if attn_mask.dtype == torch.bool:` converts a native bool mask to the
  additive `0`/`-inf` form before the causal bias is merged, instead of letting `bool + float` promote to `float32` and
  corrupt the mask.
  - **comment accuracy**: reworded to note that supplying a mask rules out SDPA's flash backend and uses the
  memory-efficient kernel instead.
  - **causal branch**: kept it as defensive code rather than deleting, and added a note that no in-tree caller passes a
  mask with `causal=True`. This keeps the PR self-contained and independent of #1050's merge order; once #1050 lands and
  sets `causal=False`, the branch just becomes unreachable.

  `ruff check` and `ruff format --check` both pass. Numerically verified float-mask and bool-mask × `causal=True` against
  a manual `softmax(scores + mask + causal_bias) @ v` reference (max abs diff ~1e-7).

**Review by Coach257 — APPROVED** [review]
